### PR TITLE
🚯 No trigger archiving and restoring

### PR DESCRIPTION
**Pull request by rowanseymour**
Turns `Trigger` into a thing that we release>delete rather than archive>delete - same for `Schedule` since those can be attached to a trigger. A large part of this PR is wanting to pass `user` to `Trigger.release` and needing then to pass to all the other things that can call that.

Captura de Pantalla 2021-06-28 a la(s) 14 58 23
Captura de Pantalla 2021-06-28 a la(s) 14 59 20

Currently this means for imports that we just release anything that conflicts.. can discuss whether we need to give users more warning about what will be replaced other whether we just need to add triggers to the existing warning..

Captura de Pantalla 2021-06-28 a la(s) 15 34 50

### CODE REVIEW

File: temba/schedules/models.py [hunk 1]
```diff
@@ -212,5 +213,19 @@ def _day_of_week(d):
         """
         return Schedule.DAYS_OF_WEEK_OFFSET[d.weekday()]
 
+    def release(self, user):
```
Comment (rowanseymour) [inline temba/schedules/models.py:216]: in the case where we're releasing a trigger with a schedule we need to also release the schedule

File: temba/schedules/models.py [hunk 2]
```diff
@@ -212,5 +213,19 @@ def _day_of_week(d):
...
         return f'Schedule[id={self.id} repeat="{self.get_display()}" next={str(self.next_fire)}]'
+
+    class Meta:
+        indexes = [
+            Index(
+                name="schedules_next_fire_active",
+                fields=["next_fire"],
+                condition=Q(is_active=True, next_fire__isnull=False),
+            )
+        ]
```
Comment (rowanseymour) [inline temba/schedules/models.py:231]: I've checked this is used by mailroom

File: temba/flows/models.py [hunk 3]
```diff
@@ -593,21 +592,22 @@ def deps_of_type(type_name):
...
         # queue mailroom to interrupt sessions where contact is currently in this flow
         mailroom.queue_interrupt(self.org, flow=self)
 
-        # archive our triggers as well
-        from temba.triggers.models import Trigger
-
-        Trigger.objects.filter(flow=self).update(is_archived=True)
+        # and release any triggers that point to this flow
+        for trigger in self.triggers.all():
+            trigger.release(user)
```
Comment (norkans7) [inline temba/flows/models.py:605]: Now that triggers will disappear completely, do we want to notify the user about that behavior?

Comment (norkans7) [reply temba/flows/models.py:605]: when archiving a flow

Comment (rowanseymour) [reply temba/flows/models.py:605]: Yeah let's chat today about what extra UI stuff we need here

**Pull request closed without merging**